server/src/domain/system-config/system-config.core.ts

```diff
@@ -91,11 +97,17 @@ export class SystemConfigCore {
   }
 
   public async getConfig() {
-    const overrides = await this.repository.load();
-    const config: DeepPartial<SystemConfig> = {};
-    for (const { key, value } of overrides) {
-      // set via dot notation
-      _.set(config, key, value);
+    let config: DeepPartial<SystemConfig> = {};
+    if (process.env.CONFIG_FILE) {
+      const file = await this.storageRepository.readFile();
+      config = JSON.parse(file);
```

Comment:
We should probably validate the contents of the config file in some way and apply them on top of the defaults, rather than just blindly loading and accepting it. At the moment it would be possible to set unknown/unsupported values, and to neglect setting values that are required.

Reply:
In system-config.core.ts the updateConfig function uses some validators that it gets injected. I guess I could use these, but I cannot find, how they are being injected and how I can get them? addValidator is only called by tests?

Reply:
I haven't tested this, but I believe you can convert the object to a class and call `class-validator`'s `validate` in `readConfigFile`:

```suggestion
        const config = plainToClass(SystemConfigDto, JSON.parse(await fs.readFile(path, 'utf-8')))
        await validate(config, { forbidNonWhitelisted: true, forbidUnknownValues: true })
```

Reply:
Unfortunately, this does not validate invalid numbers for instance. Setting a wrong resolution for e.g. Thumbnails still yields an empty field in the UI but no error.

Reply:
I've just seen that valid values in selects are currently hardcoded in the UI. So I'd assume the specific values aren't being verified at all? Maybe the backend should verify them and provide endpoints to fetch valid options? 'Cause technically this isn't something the frontend should take care of, especially because it would yield duplicate code.

Reply:
The server rejects the updated config if any of the options fail the validation decorators in the DTOs. This is done in the system config controller where there's a `@UseValidation` decorator telling Nest.js to validate requests and responses. 

But some things essentially aren't validated since they're strings.

Reply:
Yeah, I got that. But I suppose values like resolution could use `@IsIn()` with a list of valid options. Wouldn't this be even tighter validation. This list could then be exposed to the frontend as well, so that the select options can be generated from it.

Reply:
Yeah, adding validation for that would be a good change. I've been slowly migrating those settings into explicit enums but never got around to changing those.

Generating the options from the type would be elegant, but it removes some flexibility. For instance, displayed text isn't always the same as the value, and there's no guarantee that all options will always be user-facing.

Reply:
I feel like the config file and the UI should always have feature parity and, since the config file would be validated against all possible values, they should all be accessible from the UI, too.

Nevertheless, for now, this suffices as validation and we pick this up later on?

Reply:
Agreed, I think this is fine.